### PR TITLE
RUBY-2194 Provide forking guidance relevant to the driver


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/RUBY-2194

File: docs/tutorials/ruby-driver-create-client.txt [hunk 1]
```diff
@@ -1129,55 +1129,102 @@ Note that when creating a client using the `block syntax <#block-syntax>`_ descr
...
+.. note::
+  
+  Applications using Mongoid should follow `Mongoid's forking guidance
+  <https://docs.mongodb.com/mongoid/current/tutorials/mongoid-configuration/#usage-with-forking-servers>`_.
+  The guidance and sample code below is provided for applications using the
+  Ruby driver directly.
+
+When using the Mongo Ruby driver in a Web application with a forking web server
+such as Unicorn, Puma or Passenger, or when the application otherwise forks,
+each process should generally each have their own Mongo::Client instances.
```
Comment (alexbevi) [inline docs/tutorials/ruby-driver-create-client.txt:1141]: Mongo::Client should be formatted -> ``Mongo::Client``

File: docs/tutorials/ruby-driver-create-client.txt [hunk 2]
```diff
@@ -1129,55 +1129,102 @@ Note that when creating a client using the `block syntax <#block-syntax>`_ descr
...
+Web servers generally provide hooks that can be used by applications to
+perform actions when the worker processes are forked. The recommended hooks
+to use are:
+
+- For Puma, ``before_fork`` to close clients in the parent process and
+  ``on_worker_boot`` to reconnect in the child processes.
+- For Unicorn, ``before_fork`` to close clients in the parent process and
+  ``after_fork`` to reconnect clients in the child processes.
+- For Passenger, ``starting_worker_process`` to reconnect clients in the
+  child processes (Passenger does not appear to have a pre-fork hook).
```
Comment (alexbevi) [inline docs/tutorials/ruby-driver-create-client.txt:1220]: Is it worth linking to the documentation for each hook?

File: docs/tutorials/ruby-driver-create-client.txt [hunk 3]
```diff
@@ -1129,55 +1129,102 @@ Note that when creating a client using the `block syntax <#block-syntax>`_ descr
...
 
-It is recommended to not create any Mongo::Client instances prior to the fork.
-If the parent process needs to perform operations on the MongoDB database,
-reset the client in an ``after_fork`` handler which is defined in
-``unicorn.rb``:
+The driver attempts to detect client use from forked processes and 
+reestablish network connections when such use is detected, alleviating
+the issue of file descriptor sharing.
+
+If both parent and child processes both need to perform MongoDB operations,
```
Comment (egiurleo) [inline docs/tutorials/ruby-driver-create-client.txt:1153]: There are two "both"s in this clause

File: docs/tutorials/ruby-driver-create-client.txt (hunk 2 above)
Comment (egiurleo) [reply docs/tutorials/ruby-driver-create-client.txt:1220]: I agree that might be helpful

**Pull request merged**